hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/block/BlockStreamPoller.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.importer.downloader.block;
+
+import com.google.common.base.Stopwatch;
+import com.hedera.mirror.common.domain.transaction.BlockFile;
+import com.hedera.mirror.common.domain.transaction.RecordFile;
+import com.hedera.mirror.importer.addressbook.ConsensusNode;
+import com.hedera.mirror.importer.addressbook.ConsensusNodeService;
+import com.hedera.mirror.importer.domain.StreamFilename;
+import com.hedera.mirror.importer.downloader.CommonDownloaderProperties;
+import com.hedera.mirror.importer.downloader.StreamPoller;
+import com.hedera.mirror.importer.downloader.provider.StreamFileProvider;
+import com.hedera.mirror.importer.leader.Leader;
+import com.hedera.mirror.importer.reader.block.BlockFileReader;
+import com.hedera.mirror.importer.repository.RecordFileRepository;
+import com.hedera.mirror.importer.util.Utility;
+import jakarta.inject.Named;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import lombok.CustomLog;
+import lombok.RequiredArgsConstructor;
+import org.springframework.scheduling.annotation.Scheduled;
+
+@CustomLog
+@Named
+@RequiredArgsConstructor
+final class BlockStreamPoller implements StreamPoller {
+
+    private static final Optional<Long> PRE_GENESIS = Optional.of(-1L);
+
+    private final BlockFileReader blockFileReader;
+    private final BlockStreamVerifier blockStreamVerifier;
+    private final CommonDownloaderProperties commonDownloaderProperties;
+    private final ConsensusNodeService consensusNodeService;
+    private final AtomicReference<Optional<Long>> lastBlockNumber = new AtomicReference<>(Optional.empty());
+    private final BlockPollerProperties properties;
+    private final RecordFileRepository recordFileRepository;
+    private final StreamFileProvider streamFileProvider;
+
+    @Override
+    @Leader
+    @Scheduled(fixedDelayString = "#{@blockPollerProperties.getFrequency().toMillis()}")
+    public void poll() {
+        if (!properties.isEnabled()) {
+            return;
+        }
+
+        long blockNumber = getNextBlockNumber();
+        String filename = BlockFile.getBlockStreamFilename(blockNumber);
+        var nodes = getRandomizedNodes();
+        var stopwatch = Stopwatch.createStarted();
+        var streamFilename = StreamFilename.from(filename);
```

Comment:
Maybe we should add an overloaded `from(long)` to `Streamfilename`.

```suggestion
        var nodes = getRandomizedNodes();
        var stopwatch = Stopwatch.createStarted();
        var streamFilename = StreamFilename.from(blockNumber);
```